python/mp/rosary.py

```diff
@@ -53,6 +56,72 @@ def update(self):
         self.osc_client.send(msg.build())
 
 
+class PowerModeUpdater:
+    """
+    Create this in a separate thread to read from file on init,
+    and every 10 seconds both send an OSC message out, and rewrite to file
+    """
+
+    def __init__(self, osc_client=None):
+
+        self.osc_client = osc_client
+        # The file will live in the same directory as this file
+        self.backup_file_path = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                             "low_power_mode.json")
+        self.low_power_mode = self.read_from_disk()
+
+
+    def read_from_disk(self):
+
+        low_power_mode = False
+
+        try:
+            with open(self.backup_file_path, "r") as fp:
+                power_mode_parsed = json.loads(fp.read())
+                low_power_mode = power_mode_parsed.get("low_power_mode", False)
+                print("Loaded power mode from file, low power mode is: {}".format("ON" if low_power_mode else "OFF"))
+
+        # Whether the file was not found, mis-formatted, or corrupted,
+        # just continue to assume normal power mode
+        except Exception as e:
+            print("Failed to load power mode file, caught: {}".format(e))
+
+        return low_power_mode
+
+
+    def write_to_disk(self):
+
+        power_mode_jsonified = json.dumps({
+            "low_power_mode": self.low_power_mode,
+            "note": "If you're editing this, make sure 'true' / 'false' is lowercased, no quotes!"
+        })
+
+        # When all's said and done, write the value to disk
+        with open(self.backup_file_path, "w") as fp:
+            fp.write(power_mode_jsonified)
+
+
+    def update(self, low_power_mode=False):
+
+        # We could possibly only call this
+        self.low_power_mode = low_power_mode
+
+        msg = osc_message_builder.OscMessageBuilder(address = "/led/xform")
+        msg.add_arg("power")                    # I don't know
+        msg.add_arg(int(0))                     # what these two do
+        msg.add_arg(float(3.0))                 # But I get this one!
+
+        for i in range(3):
+            if low_power_mode:
+                msg.add_arg(1.0)
+            else:
+                msg.add_arg(.75)
+
+        self.osc_client.send(msg.build())
+
+        self.write_to_disk()
+
```

Comment:
Please move this into the trigger handler so its only written once per state change. I don't want to wear out the flash memory faster than necessary.